uo_sys_framework/ACREModule/cfgXEH/init/ClientPreInit.sqf

```diff
@@ -0,0 +1,410 @@
+				{
+					private _language = _x;
+					{
+						if (_language select 1 == missionNamespace getVariable [_x,""] &&  missionNamespace getVariable [_x,""] != "" ) then {
+							UO_FW_ACRE_INDEPENDENT_BABEL_LANGUAGES pushBack (_language select 0);
+						};
+					} forEach ["UO_FW_ACRE_INDEPENDENT_BABEL_Custom","UO_FW_ACRE_INDEPENDENT_BABEL_Custom2"];
+				} forEach UO_FW_CURRENTBABEL_LANGUAGES;
+
+				UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES = [];
+				if (UO_FW_ACRE_CIVILIAN_BABEL_EN) then {UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack "en"; };
+				if (UO_FW_ACRE_CIVILIAN_BABEL_RU) then {	UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack "ru"; };
+				if (UO_FW_ACRE_CIVILIAN_BABEL_FR) then {UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack "fr"; };
+				if (UO_FW_ACRE_CIVILIAN_BABEL_AR) then {UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack "ar"; };
+				if (UO_FW_ACRE_CIVILIAN_BABEL_GK) then {	UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack "gk"; };
+				{
+					private _language = _x;
+					{
+						if (_language select 1 == missionNamespace getVariable [_x,""] &&  missionNamespace getVariable [_x,""] != "" ) then {
+							UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES pushBack (_language select 0);
+						};
+					} forEach ["UO_FW_ACRE_CIVILIAN_BABEL_Custom","UO_FW_ACRE_CIVILIAN_BABEL_Custom2"];
+				} forEach UO_FW_CURRENTBABEL_LANGUAGES;
+
+
+				if (UO_FW_ACRE_BLUFOR_BABEL_Enabled  && (side player == west)) then {
+							UO_FW_ACRE_BLUFOR_BABEL_LANGUAGES call acre_api_fnc_babelSetSpokenLanguages;
+
+							private _languages = player getVariable ["UO_FW_Languages", []];
+
+							if (count _languages > 0) then {
+
+								_languages call acre_api_fnc_babelSetSpokenLanguages;
+
+							};
+				};
+
+				if (UO_FW_ACRE_OPFOR_BABEL_Enabled  && (side player == east)) then {
+							UO_FW_ACRE_OPFOR_BABEL_LANGUAGES call acre_api_fnc_babelSetSpokenLanguages;
+
+							private _languages = player getVariable ["UO_FW_Languages", []];
+
+							if (count _languages > 0) then {
+
+								_languages call acre_api_fnc_babelSetSpokenLanguages;
+
+							};
+				};
+
+				if (UO_FW_ACRE_INDEPENDENT_BABEL_Enabled  && (side player == independent)) then {
+							UO_FW_ACRE_INDEPENDENT_BABEL_LANGUAGES call acre_api_fnc_babelSetSpokenLanguages;
+
+							private _languages = player getVariable ["UO_FW_Languages", []];
+
+							if (count _languages > 0) then {
+
+								_languages call acre_api_fnc_babelSetSpokenLanguages;
+
+							};
+				};
+
+				if (UO_FW_ACRE_CIVILIAN_BABEL_Enabled  && (side player == civilian)) then {
+							UO_FW_ACRE_CIVILIAN_BABEL_LANGUAGES call acre_api_fnc_babelSetSpokenLanguages;
+
+							private _languages = player getVariable ["UO_FW_Languages", []];
+
+							if (count _languages > 0) then {
+
+								_languages call acre_api_fnc_babelSetSpokenLanguages;
+
+							};
+				};
+
+			};
+
+			[{CBA_missionTime > 0 && {[] call acre_api_fnc_isInitialized}},
```

Comment:
Do we need these multiple waituntilandexecutes?

Reply:
Also we need to check for `UO_FW_GearReady` so these settings can overwrite gear scripts and not the other way round.

Reply:
Yes because if you add an Radio you need to check for acre again